src/main/scala/firrtl/passes/ExpandWhens.scala

```diff
@@ -90,7 +99,13 @@ object ExpandWhens extends Pass {
         case c: IsInvalid =>
           netlist(c.expr) = WInvalid
           EmptyStmt
-        case c: Attach => c
+        case Attach(_, exprs) =>
```

Comment:
What are the semantics for:

Example 1:
```
attach(a, b)
when p:
  attach(b, c)
```
Example 2:
```
when p:
  attach(d, e)
```

I'd recommend example 2 to simply ignore the when statement, so the expanded form would simply be ```attach(d,e)```.

Example 1 is confusing, but I believe its best to just define the semantics that attach statements always ignore when statements, resulting in ```attach(a,b,c)```. This still abides by the instance-replacement property (arbitrary code can get wrapped in a module and instantiated, and things will still work).